sdk/nodejs/__tests__/eventQueueAS.test.ts

```diff
@@ -0,0 +1,660 @@
+                _variation: 'variation_3'
+            },
+            key_4: {
+                _feature: 'feature_4',
+                _variation: 'variation_4'
+            }
+        }
+    }
+    const mockAxiosResponse = (obj: any): AxiosResponse => ({
+        status: 200,
+        statusText: '',
+        data: {},
+        headers: {},
+        config: {},
+        ...obj
+    })
+
+    let currentEventKey = ''
+    const initEventQueue = (envKey: string, options?: EventQueueASOptions, logger?: DVCLogger): EventQueue => {
+        getBucketingLib().setConfigData(envKey, JSON.stringify(config))
+        currentEventKey = envKey
+        return new EventQueue(logger || defaultLogger, envKey, options)
+    }
+
+    beforeAll(async () => {
+        await importBucketingLib()
+        setPlatformDataJSON()
+    })
+
+    afterEach(() => {
+        publishEvents_mock.mockReset()
+        getBucketingLib().cleanupEventQueue(currentEventKey)
+    })
+
+    it('should setup Event Queue and process events', async () => {
+        publishEvents_mock.mockResolvedValue(mockAxiosResponse({ status: 201 }))
+
+        const eventQueue = initEventQueue('envKey')
+        const user = new DVCPopulatedUser({ user_id: 'user_id' })
+        const event = { type: 'test_event' }
+        eventQueue.queueEvent(user, event)
+
+        const aggEvent = { type: EventTypes.aggVariableEvaluated, target: 'key' }
+        eventQueue.queueAggregateEvent(user, aggEvent, bucketedUserConfig)
+
+        await eventQueue.flushEvents()
+        eventQueue.cleanup()
+
+        expect(publishEvents_mock).toBeCalledWith(defaultLogger, 'envKey', [
+            {
+                user: expect.objectContaining({
+                    user_id: 'user_id',
+                    createdDate: expect.any(String),
+                    lastSeenDate: expect.any(String),
+                    platform: 'NodeJS',
+                    platformVersion: expect.any(String),
+                    sdkType: 'server',
+                    sdkVersion: expect.any(String)
+                }),
+                events: [
+                    expect.objectContaining({
+                        clientDate: expect.any(String),
+                        customType: 'test_event',
+                        date: expect.any(String),
+                        featureVars: {},
+                        type: 'customEvent',
+                        user_id: 'user_id'
+                    })
+                ]
+            },
+            {
+                user: expect.objectContaining({
+                    user_id: 'host.name',
+                    createdDate: expect.any(String),
+                    lastSeenDate: expect.any(String),
+                    platform: 'NodeJS',
+                    platformVersion: expect.any(String),
+                    sdkType: 'server',
+                    sdkVersion: expect.any(String)
+                }),
+                events: [
+                    expect.objectContaining({
```

Comment:
The tests would be a lot more succinct and easier to understand if we had a function `expectEvent` which you could pass an object into with only the fields the test actually cares about the values of. That function would `expect.any()` unspecified fields as appropriate.